sdks/java/testing/load-tests/src/main/java/org/apache/beam/sdk/loadtests/SideInputLoadTest.java

```diff
@@ -0,0 +1,243 @@
+ *
+ * <pre>
+ *   ./gradlew :sdks:java:testing:load-tests:run -PloadTest.args='
+ *    --sourceOptions={"numRecords":2000, ...}
+ *    --sideInputType=ITERABLE
+ *    --accessPercentage=1
+ *    --windowCount=200
+ * </pre>
+ */
+public class SideInputLoadTest extends LoadTest<SideInputLoadTest.Options> {
+
+  private static final String METRICS_NAMESPACE = "sideinput";
+  private static final Instant TIME = new Instant();
+
+  public SideInputLoadTest(String[] args) throws IOException {
+    super(args, Options.class, METRICS_NAMESPACE);
+  }
+
+  @Override
+  void loadTest() throws IOException {
+    Optional<SyntheticStep> syntheticStep = createStep(options.getStepOptions());
+    PCollection<KV<byte[], byte[]>> input =
+        pipeline
+            .apply(readFromSource(sourceOptions))
+            .apply(ParDo.of(new AddTimestamps()))
+            .apply("Collect start time metrics", ParDo.of(runtimeMonitor))
+            .apply(ParDo.of(new ByteMonitor(METRICS_NAMESPACE, "totalBytes.count")));
+
+    performTestWithSideInput(
+        input, SideInputMaterializationType.valueOf(options.getSideInputType()), syntheticStep);
+  }
+
+  private void performTestWithSideInput(
+      PCollection<KV<byte[], byte[]>> input,
+      SideInputMaterializationType sideInputType,
+      Optional<SyntheticStep> syntheticStep) {
+    switch (sideInputType) {
+      case ITERABLE:
+        performTestWithIterable(input, syntheticStep);
+        break;
+      case HASHMAP:
+        performTestWithHashMap(input, syntheticStep);
+        break;
+      case LIST:
+        performTestWithList(input, syntheticStep);
+        break;
+    }
+  }
+
+  private void performTestWithList(
+      PCollection<KV<byte[], byte[]>> input, Optional<SyntheticStep> syntheticStep) {
+    applyStepIfPresent(input, "Synthetic step", syntheticStep);
+    PCollectionView<List<KV<byte[], byte[]>>> sideInput = input.apply(View.asList());
+    input
+        .apply(ParDo.of(new SideInputTestWithList(sideInput)).withSideInputs(sideInput))
+        .apply("Collect end time metrics", ParDo.of(runtimeMonitor));
+  }
+
+  private void performTestWithHashMap(
+      PCollection<KV<byte[], byte[]>> input, Optional<SyntheticStep> syntheticStep) {
+    applyStepIfPresent(input, "Synthetic step", syntheticStep);
+    PCollectionView<Map<byte[], byte[]>> sideInput = input.apply(View.asMap());
+    input
+        .apply(ParDo.of(new SideInputTestWithHashMap(sideInput)).withSideInputs(sideInput))
+        .apply("Collect end time metrics", ParDo.of(runtimeMonitor));
+  }
+
+  private void performTestWithIterable(
+      PCollection<KV<byte[], byte[]>> input, Optional<SyntheticStep> syntheticStep) {
+    applyStepIfPresent(input, "Synthetic step", syntheticStep);
+    PCollectionView<Iterable<KV<byte[], byte[]>>> sideInput;
+    if (options.getWindowCount() != 0) {
+      long windowDurationMilis = sourceOptions.numRecords / options.getWindowCount();
```

Comment:
Is there a specific reason we do windowing only for iterable case?

Reply:
As I understand it, all SideInput types except for Iterable are cached, so windowing shouldn't change much in other scenarios. I can add it to other cases anyway